services/replicator/replicatorclientfactory.go

```diff
@@ -65,11 +65,12 @@ type (
 
 	// repCltFactoryImpl implements of ReplicatorClientFactory
 	repCltFactoryImpl struct {
-		AppConfig         configure.CommonAppConfig
-		logger            bark.Logger
-		ch                *tchannel.Channel
-		deploymentClients map[string][]replicator.TChanReplicator
-		lk                sync.RWMutex
+		AppConfig configure.CommonAppConfig
+		logger    bark.Logger
+		ch        *tchannel.Channel
+
+		deploymentClients   map[string][]replicator.TChanReplicator
+		deploymentClientsLk sync.RWMutex
```

Comment:
Good news: Go 1.9 has concurrent maps! https://golang.org/pkg/sync/#Map

Bad news: we can't use it yet .. since we probably want to maintain support for Go 1.8 too.